src/org/labkey/remoteapi/NetrcFileParser.java

```diff
@@ -43,55 +48,55 @@ private NetrcEntry getEntry(String netrcName, String host) throws IOException
         return getEntry(netrcFile, host);
     }
 
-    public  NetrcEntry getEntry(File netrcFile, String host) throws IOException
+    // Note: Also called by test PasswordUtil
+    public NetrcEntry getEntry(File netrcFile, String host) throws IOException
     {
-        if (!netrcFile.exists())
-            return null;
-
-        BufferedReader input = null;
+        NetrcEntry entry = null;
+        LOG.info("Attempting to access " + netrcFile);
 
-        try
+        if (netrcFile.exists())
         {
-            StringBuilder sb = new StringBuilder();
-            input = new BufferedReader(new FileReader(netrcFile, StandardCharsets.UTF_8));
-            String line;
+            LOG.info(netrcFile + " was found");
+            LOG.info("Attempting to find an entry in " + netrcFile + " for host \"" + host + "\"");
 
-            while ((line = input.readLine()) != null)
+            try (BufferedReader input = new BufferedReader(new FileReader(netrcFile, StandardCharsets.UTF_8)))
             {
-                sb.append(line).append('\n');
-            }
+                StringBuilder sb = new StringBuilder();
+                String line;
 
-            Matcher m = netrcPattern.matcher(sb);
+                while ((line = input.readLine()) != null)
+                {
+                    sb.append(line).append('\n');
+                }
 
-            int index = 0;
+                Matcher m = netrcPattern.matcher(sb);
 
-            while(m.find(index))
-            {
-                if (3 == m.groupCount() && m.group(1).equals(host))
-                    return new NetrcEntry(m.group(1), m.group(2), m.group(3));
+                int index = 0;
 
-                index = m.end();
-            }
-        }
-        finally
-        {
-            try
-            {
-                if (input != null)
+                while (m.find(index))
                 {
-                    input.close();
+                    if (3 == m.groupCount() && m.group(1).equals(host))
+                    {
+                        entry = new NetrcEntry(m.group(1), m.group(2), m.group(3));
+                        break;
+                    }
+
+                    index = m.end();
                 }
             }
-            catch (IOException ioe)
-            {
-                // ignore
-            }
+
+            LOG.info("Entry corresponding to host \"" + host + "\" was " +  (null == entry ? "not " : "") + "found in " +
+                netrcFile + (null == entry ? "" : "; the login and password fields of this entry will be used to attempt authentication."));
+        }
+        else
```

Comment:
I prefer errors near the "if", sometimes these end up stacking up confusingly at the end of functions.

Reply:
Switched "not found" to up top